### PR TITLE
x/migration: add NewFSSource

### PR DESCRIPTION
**Pull request by jlevesy**
### What does this PR do?

This PR adds a new migration source constructor based on go1.16 [io/fs](https://tip.golang.org/pkg/io/fs/) package.
This open the road to using https://golang.org/pkg/embed/ for our migrations and remove gobindata generated files.

Instead of passing logger down and only logging issues, I chose to return errors explicitly and use the `Must` pattern which makes the app panic if it detects an error. As in that case it's build time issues, I do believe it's more suited. WDYT ?

Also adds PR templates and name @AlexisMontagne and @upfluence/golang as maintainers.

### Good PR checklist

- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [ ] Added/updated documentation
- [x] Properly labeled

### CODE REVIEW

File: x/migration/fs_source_test.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		ctx = context.Background()
+		fs  = fstest.MapFS{
+			"3_final.down.sql":      &fstest.MapFile{Data: []byte("bar")},
+			"2_initial.up.postgres": &fstest.MapFile{Data: []byte("foo")},
+			"3_final.up.sql":        &fstest.MapFile{Data: []byte("bar")},
+		}
+	)
+
+	s, err := NewFSSource(fs, ".")
+	require.NoError(t, err)
```
Comment (Sypheos) [inline x/migration/fs_source_test.go:26]: Can you make a test with  an error now that you are returning one :smile:

**Review by Sypheos — COMMENTED** [review]
I don't have much opinion about the `Must` pattern, if there is a migration problem it should be caught in the tests.

File: x/migration/fs_source_test.go (hunk 1 above)
Comment (jlevesy) [reply x/migration/fs_source_test.go:26]: Fair.

File: x/migration/fs_source_test.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	assert.NoError(t, err)
+	assert.False(t, ok)
+}
+
+func TestFSSource_FailsIfMigrationIsUnsupported(t *testing.T) {
+	fs := fstest.MapFS{
+		"3_final.down.sql":      &fstest.MapFile{Data: []byte("bar")},
+		"2_initial.up.postgres": &fstest.MapFile{Data: []byte("foo")},
+		"3_final.up.sql":        &fstest.MapFile{Data: []byte("bar")},
+		"bad":                   &fstest.MapFile{Data: []byte("notworking")},
```
Comment (Sypheos) [inline x/migration/fs_source_test.go]: You also need to test for L51, where two migration have the same ID but not the same name.

**Comment by jlevesy** [comment]
Here's an example of the changes we need to do on our projects https://github.com/upfluence/creator-server/compare/jl/embed-migrations

**Review by Sypheos — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 5432:5432
+
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.16.x
```
Comment (AlexisMontagne) [inline .github/workflows/ci.yml]: should we run the test as in a matrix on 1.14.x, 1.15.x and 1.16.x?

**Review by AlexisMontagne — APPROVED** [review]

**Review by Sypheos — APPROVED** [review]

**Pull request merged**